src/components/calcite-list-item/calcite-list-item.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, Element, Event, EventEmitter, Prop, h, VNode } from "@stencil/core";
+import { SLOTS, CSS } from "./resources";
+import { getSlotted } from "../../utils/dom";
+
+/**
+ * @slot actions-start - todo document.
+ * @slot content-start - todo document.
+ * @slot content-end - todo document.
+ * @slot actions-end - todo document.
+ */
+@Component({
+  tag: "calcite-list-item",
+  styleUrl: "calcite-list-item.scss",
+  shadow: true
+})
+export class CalciteListItem {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * @todo document.
+   */
+  @Prop({ reflect: true }) buttonDisabled = false;
+
+  /**
+   * @todo document.
+   */
+  @Prop() description: string;
+
+  /**
+   * @todo document.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * @todo document.
+   */
+  @Prop() label: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calciteListItemClick: EventEmitter;
```

Comment:
Can you type the payload as `void`? Otherwise, the type and doc will show `any`.